src/test/java/com/mongodb/jdbc/MongoDriverTest.java

```diff
@@ -61,16 +61,16 @@ public void execute() throws Throwable {
     void testDBURL() throws SQLException {
         MongoDriver d = new MongoDriver();
         // Should not return null or throw, even with null properties.
-        assertNotNull(d.connect(authDBURL, null));
+        assertNotNull(d.createUnvalidatedConnection(authDBURL, null));
```

Comment:
I am wondering if having this in MongoSQLMock would not work : `when(mongoConnection.isValid(anyInt())).thenReturn(true);`?
This way, no need for the special method `createUnvalidatedConnection`  just for unit test.

Reply:
This doesn't work. Invalid use of matchers. I think the problem is that we use `MongoSQLMock` in several tests.

I also think mocking should only be used for things that can't be easily handled otherwise. Adding one protected method is not a big deal.